### ISSUE TITLE
Move selection from `Snapshot` to `Commit`

### ISSUE THREAD

**Issue by oscbyspro** [oscbyspro/DiffableTextViews#154]
### Motivation

It makes more sense to `Commit` a selection than adding a selection to a `Snapshot`. Why? Because, what does `Proposal/base.selection` and `Proposal/replacement.selection` mean? Nothing. It’s nonsense, and a headache.

**Comment by oscbyspro** [comment]
This would also reduce `Snapshot` to 2 words, meaning that I would no longer need to consider copy-on-write-ing it.

**Comment by oscbyspro** [comment]
A couple of helper methods could make it quite ergonomic:

```swift
commit.select({ $0.endIndex })
commit.select({ $0.startIndex ..< $0.endIndex })
```

**Comment by oscbyspro** [comment]
Implemented in fa9a62d66fa809a48c90ac3d3764b97b4e21cdb4, pending some manual testing.
